src/Strategy/Googlesource/RepoManifest.hs

```diff
@@ -41,17 +45,58 @@ discover = walk $ \_ _ files ->
         pure WalkSkipAll
       else pure WalkContinue
 
-analyze :: (Has ReadFS sig m, Has (Error ReadFSErr) sig m, MonadFail m) => Path Rel File -> m ProjectClosureBody
-analyze file = mkProjectClosure file <$> nestedValidatedProjects file
+analyze :: (Has ReadFS sig m, Has (Error ReadFSErr) sig m, MonadFail m, Effect sig) => Path Rel File -> m ProjectClosureBody
+analyze file = mkProjectClosure file <$> nestedValidatedProjects (parent file) file
 
-nestedValidatedProjects :: (Has ReadFS sig m, Has (Error ReadFSErr) sig m, MonadFail m) => Path Rel File -> m [ValidatedProject]
-nestedValidatedProjects file = do
+nestedValidatedProjects :: (Has ReadFS sig m, Has (Error ReadFSErr) sig m, Effect sig, MonadFail m) => Path Rel Dir -> Path Rel File -> m [ValidatedProject]
+nestedValidatedProjects rootDir file = do
   manifest <- readContentsXML @RepoManifest file
-  validatedIncludedProjects <- validatedProjectsFromIncludes manifest $ parent file
-  let validatedDirectProjects = validateProjects manifest
-  case validatedDirectProjects of
-    Nothing -> fail "Error"
-    Just ps -> pure $ ps ++ validatedIncludedProjects
+  manifestWithFixedRemotes <- runError @ManifestGitConfigError $ fixRelativeRemotes manifest rootDir
+  case manifestWithFixedRemotes of
+    Left err -> fail $ "Error parsing remotes: " ++ show err
+    Right fixedManifest -> do
+      validatedIncludedProjects <- validatedProjectsFromIncludes fixedManifest (parent file) rootDir
+      let validatedDirectProjects = validateProjects fixedManifest
+      case validatedDirectProjects of
+        Nothing -> fail "Error creating validated projects"
+        Just ps -> pure $ ps ++ validatedIncludedProjects
+
+fixRelativeRemotes :: (Has ReadFS sig m, Has (Error ReadFSErr) sig m, Has (Error ManifestGitConfigError) sig m) => RepoManifest -> Path Rel Dir -> m RepoManifest
+fixRelativeRemotes manifest rootDir = do
+  let remotes = manifestRemotes manifest
+  fixedRemotes <- traverse (fixRemote rootDir) remotes
+  pure $ manifest {manifestRemotes = fixedRemotes}
+
+fixRemote :: (Has ReadFS sig m, Has (Error ReadFSErr) sig m, Has (Error ManifestGitConfigError) sig m) => Path Rel Dir -> ManifestRemote -> m ManifestRemote
+fixRemote rootDir remote = do
+  let configFile = rootDir </> $(mkRelFile "manifests.git/config")
+
+  exists <- doesFileExist configFile
+  unless exists (throwError $ MissingGitConfig $ T.pack $ show configFile)
+
+  contents <- readContentsText configFile
+
+  case parseConfig contents of
+    Left err -> throwError (GitConfigParse (T.pack (errorBundlePretty err)))
+    Right config -> do
```

Comment:
Using the great `ManifestGitConfigError` you added, let's pair on improving the implementation here (and maybe error handling across the analyzer)